www/js/diary/current.js

```diff
@@ -0,0 +1,805 @@
+      from_local_date: $scope.selectCtrl.fromDate,
+      to_local_date: $scope.selectCtrl.toDate,
+      sel_region: null
+    };
+    Logger.log("Sending data "+JSON.stringify(data));
+    return $http.post("https://e-mission.eecs.berkeley.edu/result/heatmap/incidents/local_date", data)
+    .then(function(response) {
+      if (angular.isDefined(response.data.incidents)) {
+        Logger.log("Got incidents"+response.data.incidents.length);
+        $scope.showIncidents(response.data.incidents);
+      } else {
+        Logger.log("did not find incidents in response data "+JSON.stringify(response.data));
+      }
+      $scope.stressData.isLoading = false;
+    }, function(error) {
+      Logger.log("Got error %s while trying to read heatmap data" +
+        JSON.stringify(error));
+      $scope.stressData.isLoading = false;
+    });
+  };
+
+  $scope.showIncidents = function(incidentEntries) {
+    var incidentFeatureList = incidentEntries.map(PostTripManualMarker.toGeoJSONFeature);
+    var incidentsGeojson = L.geoJson(null, {
+      pointToLayer: PostTripManualMarker.incidentMarker,
+      onEachFeature: PostTripManualMarker.displayIncident
+    });
+    incidentFeatureList.forEach(function(ival, i, array) {
+      incidentsGeojson.addData(ival);
+    });
+    var bounds = L.featureGroup([incidentsGeojson]).getBounds();
+    Logger.log("geojson bounds="+JSON.stringify(bounds));
+    $scope.stressData.layer = incidentsGeojson;
+    $scope.stressData.bounds = bounds;
+    leafletData.getMap('current').then(function(map) {
+      console.log("success, got map "+map);
+      incidentsGeojson.addTo(map);
+    })
+    .catch(function(error) {
+      console.log("error while getting map current from leafletData");
+    });
+    Logger.log("stress layer"+incidentFeatureList);
+
+  }
+
+  $scope.getIncidents();
+  
+  var _map;
+//   $scope.addfeature = function() {
+//     // $scope.lat_long();
+//     // var newFeature = L.GeoJSON.asFeature(both);
+//     // newFeature.properties.feature_type = "incident";
+//     var geojsonFeature = {
+//     "type": "Feature",
+//     "properties": {
+//         "name": "incident",
+//     },
+//     "geometry": {
+//         "type": "Point",
+//         "coordinates": both
+//     }
+// }; 
+// } 
+   
+  var addSafe = function(marker, map) {
+      marker.setStyle({color: 'green'});
+      map.addLayer(marker);
+  };
+
+  /*
+   * INTERNAL FUNCTION, not part of factory
+   *
+   * Add a suck (red, stress = 100) entry.
+   */
+
+  var addSuck = function(marker, map) {
+    marker.setStyle({color: 'red'});
+    map.addLayer(marker);
+  };
+
+    $scope.addmarker = function(map) {
+      console.log("both", both);
+    //   db.getAllSensorData($scope.config.key_data_mapping.Locations.key).then(function(result) {
+    //     both = [result[0].data.latitude, result[0].data.longitude];
+    // });
+      console.log("called addmarker with map = "+map);
+      var latlng = L.latLng(both);
```

Comment:
I am not even sure that this works. this will add an incident at `both`. But `both` is set when the screen is launched. So if you launch the screen, wait until you move and then go back to the (already running app in the background) and report an incident, it will do so at the place you were when you first launched the app.

Please describe how you tested this...